### DISCUSSION TITLE
クイズの問題をランダムで発生させたい

### DISCUSSION THREAD

**Discussion by silli-3072** [lit-iphone-course/lit-iphone-course#24] [Question-original] [answered]
２点質問です！！
Realmに登録した単語をTableViewに表示したい。その時にアートボードのオレンジの単語帳のように、日本語と韓国語を同じ列で2行にして表示したい。
✅`cell.textLabel?.numberOfLines=0`のように複数行かけるコードを書きましたがうまくいきませんでした。
✅ストーリボード上でセル内にラベルを二つおきそれぞれの内容を指定するコードも試しましたが「The japaneseLabel outlet from the WordViewController to the UILabel is invalid. Outlets cannot be connected to repeating content.」とエラーが出て解決できませんでした。

もう一点、クイズの問題についてです。クイズの問題をRealm上のデータからランダムで発生させた変数を用いて、問題を表示するラベルにRealmからの持ち出しをしようとしましたが、`let userData = realm.objects(Word.self)`この部分でエラーが出てしまいました。この部分でselfが使えないのが原因だと考え、クイズを作るためのメソッドを作りましたが今度はその中で乱数を発生させられませんでした。

これらどのように対応すればよろしいでしょうか？お手数ですが回答お願いします！

**Comment by txxxxc** [comment, marked as answer]
> ストーリボード上でセル内にラベルを二つおきそれぞれの内容を指定するコードも試しましたが「The japaneseLabel outlet from the WordViewController to the UILabel is invalid. Outlets cannot be connected to repeating content.」とエラーが出て解決できませんでした。

ここらへんが答えっぽいから一旦これ見ながら解決してみてほしい🙏
[Outlet cannot be connected to repeating content – Xcodeエラー | EasyRamble](https://easyramble.com/outlet-error-in-repeating-content.html)
[【Xcodeエラー解決】”Outlets cannot be connected to repeating content”のエラー対処方法まとめ/Swift独学マスター | Programmable](https://www.mechengjp.com/%E3%80%900%E3%81%8B%E3%82%89swift%E7%8B%AC%E5%AD%A6%E3%80%91outlet%E3%81%AB%E9%96%A2%E3%81%99%E3%82%8B%E3%82%A8%E3%83%A9%E3%83%BC%E5%AF%BE%E5%87%A6%E6%B3%95/)


> let userData = realm.objects(Word.self)この部分でエラーが出てしまいました。

どんなエラーが出たか教えてもらえるとめっちゃ嬉しい！(エラーが出た時は必ずエラーメッセージ添えることを習慣してほしいかもです👀)

**Reply by silli-3072** [reply]
エラーコードこちらです！

> Cannot use instance member 'realm' within property initializer; property initializers run before 'self' is available

テーブルビューのラベル表示、もう少しやってみます！

**Reply by nokonoko2057** [reply]
@silli-3072 

エラー出たのここかな？
https://github.com/silli-3072/LA4_TimeKorean/blob/027753ceac80834839dcefe9a438a12a77d1a0b9/TimeKorean/GameViewController.swift#L14

これを書く場所がよくないみたいです。
```swift
let userData = realm.objects(Word.self)
```

他のところと同じように、関数の中で呼んでみよう
viewWillAppearで呼ぶと毎回画面が表示されるたびにいい感じに出るよ
ライフサイクルの参考: https://ios-docs.dev/life-cycle/

**Reply by txxxxc** [reply]
UITableViewCellをカスタムする場合は、ファイルを分けて作る必要があるので下のやつ参考にしながらやってみてー！↓
[【Swift5/Xcode】カスタムセルを使用したTable View(テーブルビュー)の使い方〜初心者にわかりやすく解説〜 | satorikublog](https://satoriku.com/customcell/)

>エラーコードこちらです！
Cannot use instance member 'realm' within property initializer; property initializers run before 'self' is available

クラスを初期化する前にプロパティは使えないですよっていうやつなので、viewDidLoadの中でRealmからデータを取って上げる必要がありそう！

**Reply by txxxxc** [reply]
TODOアプリとおんなじやり方でやったらいけるとおもう！
https://github.com/silli-3072/LA4_ToDoApp/blob/main/ToDoApp/ViewController.swift

**Reply by silli-3072** [reply]
問題のランダム表示は、viewDidLoadで呼び出してあげる&乱数の発生方法を変えたら実装できました！！
テーブルビューはいただいた[記事](https://satoriku.com/customcell/)を参考にやってstep 11を書いたタイミングで写真のようにエラーがいっぱい出てきてしまいうまくできません😭
<img width="1512" alt="スクリーンショット 2022-09-16 15 06 20" src="https://user-images.githubusercontent.com/105117616/190568134-75f60aa5-c488-49f5-bac3-4f37e9b906bc.png">

[GitHub](https://github.com/silli-3072/LA4_TimeKorean)こちらです！

**Reply by txxxxc** [reply]
多分20行目のカッコ消したら治る気がする.....

**Reply by silli-3072** [reply]
めっちゃ治りました！！気が付かずすみません💦

**Reply by txxxxc** [reply]
いえいえ！全然いいよ！笑

**Comment by txxxxc** [comment]
GitHubにエラー出てるところのコードってあがってる？(リンクも添えてもらえると嬉しいかも)

**Reply by silli-3072** [reply]
上がっております！！[GitHub](https://github.com/silli-3072/LA4_TimeKorean)こちらです！

テーブルビューの方ですが、コードいろいろやってみましたがうまくいっておらず、あわせて確認していただけると幸いです！
